astrosat/utils/utils_utils.py

```diff
@@ -0,0 +1,15 @@
+def flatten_dictionary(dictionary, key="", separator="."):
+    """
+    Takes a nested dictionary and converts it to a 1D dictionary
+    (keys of nested branches are concatenated together using separator)
+    """
+
+    dictionary_items = []
+    for k, v in dictionary.items():
+        new_key = key + separator + k if key else k
+        if isinstance(v, dict):
+            dictionary_items.extend(
+                flatten_dictionary(v, key=new_key, separator=separator).items())
+        else:
+            dictionary_items.append((new_key, v))
+    return dict(dictionary_items)
```

Comment:
Haven't I seen this code elsewhere, are you now exporting the function from an app, to the library, so it can be re-used?

Reply:
Yes.  I wrote it for **a4h** and then it turned out I needed to do something very similar for **orbis**.  So I have added it to this library.  Next I will upgrade the depnediences in **orbis** & **a4h** and then remove the duplciate code from **a4h**.